src/templates/base-template.tsx

```diff
@@ -98,9 +99,10 @@ const Template = ({
       ),
     [contentOrderedList, language, filteredLanguages, version, versions],
   );
+
   useEffect(() => {
     if (language === DEFAULT_LANGUAGE || !filteredLanguages.includes(language)) {
-      const preferredLanguage = safeWindow.localStorage.getItem(PREFERRED_LANGUAGE_KEY);
+      const preferredLanguage = safeWindow.sessionStorage.getItem(PREFERRED_LANGUAGE_KEY);
```

Comment:
Could this not be an additional function exported from `page-language-context.tsx`? Ideally the outside world should not know how the context implements the language tracking (session storage, local storage, magic potion...)